data-prepper-expression/src/main/antlr/DataPrepperExpression.g4

```diff
@@ -17,16 +17,33 @@ expression
     ;
 
 stringExpression
-    : function
+    : stringExpression PLUS stringExpression
+    | function
     | jsonPointer
     | String
     ;
 
 arithmeticExpression
+    : arithmeticExpression (PLUS | SUBTRACT) arithmeticTerm
+    | arithmeticTerm
+    ;
+
+arithmeticTerm
+    : arithmeticTerm (MULTIPLY | DIVIDE) arithmeticFactor
+    | arithmeticFactor
+    ;
+
+arithmeticFactor
```

Comment:
Might this be the `arithmeticTerm`?

Then what you have as `arithmeticTerm` could be an `arithmeticFactor`.

Reply:
This page for the C grammar calls these "multiplicative_expression" and "additive_expression".

https://www.lysator.liu.se/c/ANSI-C-grammar-y.html